x/lockup/keeper/lock_test.go

```diff
@@ -45,6 +45,67 @@ func (suite *KeeperTestSuite) TestBeginUnlocking() { // test for all unlockable
 	suite.Require().NotEqual(locks[0].IsUnlocking(), false)
 }
 
+func (suite *KeeperTestSuite) TestBeginForceUnlock() {
+	// coins to lock
+	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10)}
+
+	testCases := []struct {
+		name             string
+		coins            sdk.Coins
+		unlockCoins      sdk.Coins
+		expectSameLockID bool
+	}{
+		{
+			name:             "new lock id is returned if the lock was split",
+			coins:            coins,
+			unlockCoins:      sdk.Coins{sdk.NewInt64Coin("stake", 1)},
+			expectSameLockID: false,
+		},
+		{
+			name:             "same lock id is returned if the lock was not split",
+			coins:            coins,
+			unlockCoins:      sdk.Coins{},
+			expectSameLockID: true,
+		},
+	}
+
+	for _, tc := range testCases {
+		suite.SetupTest()
+
+		// initial check
+		locks, err := suite.App.LockupKeeper.GetPeriodLocks(suite.Ctx)
+		suite.Require().NoError(err)
+		suite.Require().Len(locks, 0)
+
+		// lock coins
+		addr1 := sdk.AccAddress([]byte("addr1---------------"))
+		suite.LockTokens(addr1, tc.coins, time.Second)
+
+		// check locks
+		locks, err = suite.App.LockupKeeper.GetPeriodLocks(suite.Ctx)
+		suite.Require().NoError(err)
+
+		for _, lock := range locks {
```

Comment:
suggest checking that the locks length is not zero here. If we don't check and it happens so that 0 locks are returned, we would skip all the validations in the for loop completely